Improve some import handling

dopplershift opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/Unidata/MetPy/blob/main/CONTRIBUTING.md
-->

#### Description Of Changes
* Reduce import load of `metpy.io.metar`. This is currently causing importing `metpy.io` to trigger imports of `metpy.calc` and `metpy.plots`, which can add almost half a second to the total import and currently triggers warnings when Cartopy isn't installed
* Move handling of our Cartopy map features to only warn (when Cartopy isn't installed) upon *access* of the relevant attributes. This uses the module-level `__getattr__()` we have available now that we're only supporting Python >=3.7.

<!--
Feel free to remove check-list items aren't relevant to your change

Please use keywords (e.g., Fixes, Closes) to create link to the issues or pull
requests you resolved, so that they will automatically be closed when your pull
request is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->

#### Checklist

- [x] Closes #1901

dcamron reviewed (approved):
Good to go. If it doesn't matter for directly-accessed `plots.__dir__()` to be sorted, you can drop that. `dir(plots)` and iPython tab-completion already return sorted. Don't envision it hurting to leave in.

dopplershift commented:
Thanks for catching that, it was an erroneous copy from [this PEP](https://www.python.org/dev/peps/pep-0562/). I fixed, and addressed a couple things the bots pointed out. Let's see now if the failure path for `__getattr__()` is still covered...

dopplershift commented:
This should be fully tested now and ready for final review.

dopplershift commented:
Ugh, failed on build without CartoPy installed. Fixed.

dopplershift commented:
While looking elsewhere, I was inspired and decided that instead of logging a warning about needing Cartopy, it should just be part of the `AttributeError` we would raise.
